include/seqan3/alphabet/quality/phred42.hpp

```diff
@@ -172,9 +173,8 @@ struct phred42
      *
      * Constant.
      */
-    constexpr phred42 & assign_char(char_type const c)
+    constexpr phred42 & assign_char(char_type const c) noexcept
     {
-        assert(c >= offset_char && c < offset_char + max_value_size);
```

Comment:
Can we add this again? This should not have any conflict with noexcept, since it does not throw but terminate.